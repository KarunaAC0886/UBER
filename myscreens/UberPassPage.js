import { View, Text, StatusBar, Image, TouchableOpacity } from 'react-native'
import React from 'react'
import BackNavigator from './BackNavigator'
import styles from './styles'
import  {FontAwesome5}  from '@expo/vector-icons'

const UberPassPage = () => {
  return (
    <View style={styles.Uberpasspg_main_page}>
      <BackNavigator titleName="Uber Pass" navpath="account"/>
      <View style={styles.UberPasspg_cont}>
        <View style={styles.UberPasspg_frst_cont}>
          <TouchableOpacity>
            <Text style={styles.UberPasspg_frst_uptxt}>Uber Pass</Text>
            <Text style={styles.UberPasspg_frst_uprate}>1 week free - $24.99/mo</Text>
            <Text style={styles.UberPasspg_frst_updtrt}>$24.99/mo 1 week free</Text>
          </TouchableOpacity>
          <View>
            <Text style={styles.UberPasspg_frst_upmembersp}>Go more places and get more local favorites,
              all with one membership</Text>
          </View>
        </View>
        <View style={styles.UberPasspg_scnd_cont}>
          <TouchableOpacity style={styles.UberPasspg_scnd_cntr1}>
            <Image source={require("../assets/savingsEveryride.png")} />
            <View style={styles.UberPasspg_scnd_cntr_contnt_box}>
              <Text style={styles.UberPasspg_scnd_cntr_contnt1}>Savings on every ride</Text>
              <Text style={styles.UberPasspg_scnd_cntr_contnt2}>Uber Pass has you covered-10% off
                every UberX,UberL and Comfort ride,
                15% off every black,Premier, and SUV
                ride in the US.</Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity style={styles.UberPasspg_scnd_cntr2}>
            <Image source={require("../assets/freeDeliver.png")} />
            <View style={styles.UberPasspg_scnd_cntr_contnt_box}>
              <Text style={styles.UberPasspg_scnd_cntr_contnt1}>Free delivery on Uber Eats</Text>
              <Text style={styles.UberPasspg_scnd_cntr_contnt2}>Get a $0 Delivery Fee + 5% off orders
                over $15. Look for the tickets to save at
                more than 13000 restaurants in your
                area.</Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity style={styles.UberPasspg_scnd_cntr3}>
            <Image source={require("../assets/Cancelanytime.png")} />
            <View style={styles.UberPasspg_scnd_cntr_contnt_box}>
              <Text style={styles.UberPasspg_scnd_cntr_contnt1}>Cancel anytime</Text>
              <Text style={styles.UberPasspg_scnd_cntr_contnt2}>Cancel your subscription anytime-no
                penalties or fees.</Text>
            </View>
          </TouchableOpacity>
        </View>
        <View style={styles.UberPasspg_trd_cont}>
          <TouchableOpacity style={styles.UberPasspg_trd_lmbtn}><Text style={styles.UberPasspg_trd_lmlbl}>Learn More</Text><FontAwesome5 style={styles.UberPasspg_trd_lmlbl} name='greater-than'/></TouchableOpacity>
        </View>
        <View style={styles.UberPasspg_frth_cont}>
          <TouchableOpacity style={styles.UberPasspg_frth_btn}><Text style={styles.UberPasspg_frth_wfree}>Get 1 week free</Text></TouchableOpacity>
        </View>
      </View>
      <StatusBar />
    </View>
  )
}

export default UberPassPage