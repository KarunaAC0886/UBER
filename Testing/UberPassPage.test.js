import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import UberPassPage from '../myscreens/UberPassPage';

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: 'FontAwesome5', 
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));
jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));
describe('UberPassPage', () => {
  it('UberPassPage page renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<UberPassPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});