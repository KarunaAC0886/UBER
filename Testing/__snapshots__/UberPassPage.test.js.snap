// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UberPassPage UberPassPage page renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1A1A1A",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "height": "13%",
        "justifyContent": "space-between",
        "paddingBottom": "5%",
        "paddingHorizontal": "6%",
        "paddingVertical": "2%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <FontAwesome5
        name="arrow-left"
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Roboto",
          "fontSize": 36,
        }
      }
    >
      Uber Pass
    </Text>
  </View>
  <View
    style={
      {
        "flex": 0,
        "flexDirection": "column",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flex": 0,
          "flexDirection": "column",
          "height": "25%",
          "justifyContent": "space-around",
          "paddingHorizontal": "6%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 48,
              "fontWeight": "bold",
            }
          }
        >
          Uber Pass
        </Text>
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 24,
            }
          }
        >
          1 week free - $24.99/mo
        </Text>
        <Text
          style={
            {
              "color": "#979797",
              "fontSize": 18,
            }
          }
        >
          $24.99/mo 1 week free
        </Text>
      </View>
      <View>
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 17,
            }
          }
        >
          Go more places and get more local favorites, all with one membership
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#979797",
          "borderBottomWidth": 0.2,
          "height": "40%",
          "justifyContent": "space-around",
          "paddingHorizontal": "6%",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "30%",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/savingsEveryride.png",
            }
          }
        />
        <View
          style={
            {
              "flex": 0,
              "flexDirection": "column",
              "height": "100%",
              "paddingLeft": "4%",
              "width": "85%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Savings on every ride
          </Text>
          <Text
            style={
              {
                "color": "#DADADA",
                "fontSize": 14,
                "fontWeight": "400",
                "paddingTop": "1%",
              }
            }
          >
            Uber Pass has you covered-10% off every UberX,UberL and Comfort ride, 15% off every black,Premier, and SUV ride in the US.
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "30%",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/freeDeliver.png",
            }
          }
        />
        <View
          style={
            {
              "flex": 0,
              "flexDirection": "column",
              "height": "100%",
              "paddingLeft": "4%",
              "width": "85%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Free delivery on Uber Eats
          </Text>
          <Text
            style={
              {
                "color": "#DADADA",
                "fontSize": 14,
                "fontWeight": "400",
                "paddingTop": "1%",
              }
            }
          >
            Get a $0 Delivery Fee + 5% off orders over $15. Look for the tickets to save at more than 13000 restaurants in your area.
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "height": "30%",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/Cancelanytime.png",
            }
          }
        />
        <View
          style={
            {
              "flex": 0,
              "flexDirection": "column",
              "height": "100%",
              "paddingLeft": "4%",
              "width": "85%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Cancel anytime
          </Text>
          <Text
            style={
              {
                "color": "#DADADA",
                "fontSize": 14,
                "fontWeight": "400",
                "paddingTop": "1%",
              }
            }
          >
            Cancel your subscription anytime-no penalties or fees.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#979797",
          "borderBottomWidth": 0.2,
          "height": "10%",
          "justifyContent": "center",
          "paddingHorizontal": "6%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 18,
            }
          }
        >
          Learn More
        </Text>
        <FontAwesome5
          name="greater-than"
          style={
            {
              "color": "#ffffff",
              "fontSize": 18,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "height": "12%",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#EDF6FF",
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 24,
            }
          }
        >
          Get 1 week free
        </Text>
      </View>
    </View>
  </View>
</View>
`;
